frontend/packages/aquascope-editor/src/lib.ts

```diff
@@ -129,11 +135,44 @@ let parseWithDelimiters = (
   return { type: "ok", code: outputCode.join(""), ranges };
 };
 
+let parseWithMatch = (code: string, delimiter: string): ParseResult => {
+  let r = new RegExp(`^${_.escapeRegExp(delimiter)}`);
+  let makeCheck = (s: string) => {
+    let match = s.match(r);
+    return match ? match[0].length : null;
+  };
+  let ranges: Range[] = [];
+  let outputCode: string[] = [];
+  let index = 0;
+  let i = 0;
+  while (i < code.length) {
+    let n = makeCheck(code.substring(i));
+    if (n) {
+      ranges.push({
+        char_start: index,
+        char_end: index + n,
+        byte_start: 0,
+        byte_end: 0,
+        filename: "",
+      });
+      i += n;
+      continue;
+    }
+    outputCode.push(code[i]);
+    i += 1;
+    index += 1;
+  }
+
+  return { type: "ok", code: outputCode.join(""), ranges };
+};
+
 let buildStepperConfig = (config: StepperConfig) => {
   config.focusedCharPos = [];
   config.focusedPaths = new Map();
   return (toks: string[], range: Range): boolean => {
-    let hasTag = toks.length >= 5 && toks.slice(0, 5).join("") === "step:";
+    let s = "step:";
+    let n = s.length;
+    let hasTag = toks.length >= 5 && toks.slice(0, n).join("") == s;
```

Comment:
Ah I meant to write `toks.length >= n`. Also `toks.slice(5)` should be `toks.slice(n)` below.